From 3ba18d5d84738c663e16d7699342102e30e2fbc5 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.dev>
Date: Thu, 8 Feb 2024 19:25:55 +0100
Subject: [PATCH] meson: don't hard-require glslang internal dependencies

They apparently like to randomly remove them. Just mark them as not
required.

Fixes: https://code.videolan.org/videolan/libplacebo/-/issues/321
---
 src/glsl/meson.build | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/glsl/meson.build b/src/glsl/meson.build
index 5cebfb891..5a8819605 100644
--- a/src/glsl/meson.build
+++ b/src/glsl/meson.build
@@ -42,10 +42,10 @@ elif not glslang_req.disabled()
     if static
       glslang_deps += [
         # Always required for static linking
-        cxx.find_library('MachineIndependent', required: true, static: true),
-        cxx.find_library('OSDependent',        required: true, static: true),
-        cxx.find_library('OGLCompiler',        required: true, static: true),
-        cxx.find_library('GenericCodeGen',     required: true, static: true),
+        cxx.find_library('MachineIndependent', required: false, static: true),
+        cxx.find_library('OSDependent',        required: false, static: true),
+        cxx.find_library('OGLCompiler',        required: false, static: true),
+        cxx.find_library('GenericCodeGen',     required: false, static: true),
         # SPIRV-Tools are required only if optimizer is enabled in glslang build
         cxx.find_library('SPIRV-Tools',        required: false, static: true),
         cxx.find_library('SPIRV-Tools-opt',    required: false, static: true),
-- 
GitLab

